src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -328,6 +328,8 @@ public static void AddCommonServices(IServiceCollection services)
             // to be careful with caching, etc. E.g. these services will get
             // initially created in placeholder mode and live on through the
             // specialized app.
+            services.AddSingleton<HostNameProvider>();
```

Comment:
My changes work when running the host normally, but some CI tests are failing for some reason seemingly due to the service changes made. Our tests have some complicated logic where they build the host in a custom way not using production builder codepaths. I'm debugging this.